timemachine/cpp/src/kernels/k_neighborlist.cuh

```diff
@@ -22,71 +21,95 @@ void __global__ k_find_block_bounds(
-
-    const RealType bx = box[0 * 3 + 0];
-    const RealType by = box[1 * 3 + 1];
-    const RealType bz = box[2 * 3 + 2];
-
-    const RealType inv_bx = 1 / bx;
-    const RealType inv_by = 1 / by;
-    const RealType inv_bz = 1 / bz;
-    RealType pos_x = coords[atom_idx * 3 + 0];
-    RealType pos_y = coords[atom_idx * 3 + 1];
-    RealType pos_z = coords[atom_idx * 3 + 2];
-
-    RealType minPos_x = pos_x;
-    RealType minPos_y = pos_y;
-    RealType minPos_z = pos_z;
-
-    RealType maxPos_x = pos_x;
-    RealType maxPos_y = pos_y;
-    RealType maxPos_z = pos_z;
-
-    const int last = min(row_idx + TILE_SIZE, num_indices);
-    for (int i = row_idx + 1; i < last; i++) {
-        atom_idx = row_idxs[i];
+
+    if (row_idx < num_indices) {
+        int atom_idx = row_idxs[row_idx];
+
         pos_x = coords[atom_idx * 3 + 0];
         pos_y = coords[atom_idx * 3 + 1];
         pos_z = coords[atom_idx * 3 + 2];
-        // Build up center over time, and recenter before computing
-        // min and max, to reduce overall size of box thanks to accounting
-        // for periodic boundary conditions
-        RealType center_x = static_cast<RealType>(0.5) * (maxPos_x + minPos_x);
-        RealType center_y = static_cast<RealType>(0.5) * (maxPos_y + minPos_y);
-        RealType center_z = static_cast<RealType>(0.5) * (maxPos_z + minPos_z);
-        pos_x -= bx * nearbyint((pos_x - center_x) * inv_bx);
-        pos_y -= by * nearbyint((pos_y - center_y) * inv_by);
-        pos_z -= bz * nearbyint((pos_z - center_z) * inv_bz);
-        minPos_x = min(minPos_x, pos_x);
-        minPos_y = min(minPos_y, pos_y);
-        minPos_z = min(minPos_z, pos_z);
-
-        maxPos_x = max(maxPos_x, pos_x);
-        maxPos_y = max(maxPos_y, pos_y);
-        maxPos_z = max(maxPos_z, pos_z);
-    }
 
-    block_bounds_ctr[tile_idx * 3 + 0] = static_cast<RealType>(0.5) * (maxPos_x + minPos_x);
-    block_bounds_ctr[tile_idx * 3 + 1] = static_cast<RealType>(0.5) * (maxPos_y + minPos_y);
-    block_bounds_ctr[tile_idx * 3 + 2] = static_cast<RealType>(0.5) * (maxPos_z + minPos_z);
+        min_pos_x = pos_x;
+        min_pos_y = pos_y;
+        min_pos_z = pos_z;
 
-    block_bounds_ext[tile_idx * 3 + 0] = static_cast<RealType>(0.5) * (maxPos_x - minPos_x);
-    block_bounds_ext[tile_idx * 3 + 1] = static_cast<RealType>(0.5) * (maxPos_y - minPos_y);
-    block_bounds_ext[tile_idx * 3 + 2] = static_cast<RealType>(0.5) * (maxPos_z - minPos_z);
+        max_pos_x = min_pos_x;
+        max_pos_y = min_pos_y;
+        max_pos_z = min_pos_z;
+    }
 
-    // Reset the ixn count
-    if (tile_idx == 0) {
-        ixn_count[0] = 0;
+    // Only the first thread in each warp computes the min/max of the bounding box
+    bool compute_bounds = threadIdx.x % WARP_SIZE == 0;
+
+    // Build up center over time, and recenter before computing
+    // min and max, to reduce overall size of box thanks to accounting
+    // for periodic boundary conditions
```

Comment:
nit: typo "thanks to accounting for"

Reply:
The `for` I think you are saying is missing is on the next line. Not clear what the typo you are referring to is

Reply:
Ah, sorry. Parsed that as "thanks to" and "accounting for" being redundant. Can ignore.